wasm/src/value.h

```diff
@@ -43,13 +44,19 @@ typedef struct
     unsigned char free; // if set 's' is not a reference and should be freed
 } opa_number_ref_t;
 
+typedef struct
+{
+    mpd_t *d;
+    unsigned char free;
+} opa_number_mpd_t;
+
 typedef struct
 {
     opa_value hdr;
     unsigned char repr;
     union {
         long long i;
-        double f;
+        opa_number_mpd_t mpd;
```

Comment:
I've been contemplating getting rid of opa_number_ref, too. We'll end up converting it to mpd anyways. Maybe later.